### ISSUE TITLE
route53_facts will fail when using hosted_zone_id

### ISSUE THREAD

**Issue by stepanstipl** [ansible/ansible-modules-extras#1465]
Following Ansible task will fail with boto error:

```
    - name: Get zone details
      route53_facts:
        query: hosted_zone
        hosted_zone_id: "XYZ"
        hosted_zone_method: details
      register: res_dns_ns
```

Error:

```
An exception occurred during task execution. The full traceback is:
Traceback (most recent call last):
  File "/Users/stepan/.ansible/tmp/ansible-tmp-1452188296.23-178912160931026/route53_facts", line 2637, in <module>
    main()
  File "/Users/stepan/.ansible/tmp/ansible-tmp-1452188296.23-178912160931026/route53_facts", line 426, in main
    results = invocations[module.params.get('query')](route53, module)
  File "/Users/stepan/.ansible/tmp/ansible-tmp-1452188296.23-178912160931026/route53_facts", line 356, in hosted_zone_details
    results = hosted_zone_invocations[module.params.get('hosted_zone_method')](client, module)
  File "/Users/stepan/.ansible/tmp/ansible-tmp-1452188296.23-178912160931026/route53_facts", line 186, in get_hosted_zone
    results = client.get_hosted_zone(**params)
  File "/opt/boxen/homebrew/lib/python2.7/site-packages/botocore/client.py", line 310, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/opt/boxen/homebrew/lib/python2.7/site-packages/botocore/client.py", line 383, in _make_api_call
    api_params, operation_model, context=request_context)
  File "/opt/boxen/homebrew/lib/python2.7/site-packages/botocore/client.py", line 436, in _convert_to_request_dict
    api_params, operation_model)
  File "/opt/boxen/homebrew/lib/python2.7/site-packages/botocore/validate.py", line 273, in serialize_to_request
    raise ParamValidationError(report=report.generate_report())
botocore.exceptions.ParamValidationError: Parameter validation failed:
Missing required parameter in input: "Id"
Unknown parameter in input: "HostedZoneId", must be one of: Id

fatal: [localhost]: FAILED! => {"changed": false, "failed": true, "invocation": {"module_name": "route53_facts"}, "parsed": false}
```

Ansible: ansible 2.1.0 (devel 11b55be5bb)
Boto: 1.2.3

I've looked into source code and seems that boto is expecting the parameter to be called just `Id`, instead of `HostedZoneId`. I'll open a PR with a fix.

**Comment by robynbergeron** [comment]
[module: cloud/amazon/route53_facts.py]

**Comment by robynbergeron** [comment]
Hi @stepanstipl -- thanks for submitting this bug report, and your accompanying PR. 

@Etherdaemon -- thanks in advance for taking a look at this bug report and for your ongoing work in maintaining this module. Looks like there is a PR associated to fix this, and you already gave it the thumbs-up, so hopefully this should be fixed! :)

**Comment by mscherer** [comment]
Close this, since 8f06acfa92d seems to fix it.

**Comment by mscherer** [comment]
Reopen, didn't the v2 milestone.

**Comment by bcoca** [comment]
@mscherer ????

**Comment by mscherer** [comment]
"didn't see". I was doing some bug triaging, but since there is a specific milestone and this commit didn't seems to have been backported to the 2.0 branch, I reopened it as I supposed the milestone is a indicator of "need to be backported".

**Comment by ansibot** [comment]
@Etherdaemon, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@Etherdaemon, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@Etherdaemon, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by stepanstipl** [comment]
@mscherer I believe this was fixed by PR #1466 that has been merged into devel (hope you don't mind me closing this guys)
